gmso/formats/mcf.py

```diff
@@ -209,53 +216,54 @@ def _write_atom_information(mcf, top, in_ring):
         Topology object
     in_ring : list
         Boolean for each atom idx True if atom belongs to a ring
-
     """
-    names = [site.name for site in top.sites]
-    types = [site.atom_type.name for site in top.sites]
+    # Based upon Cassandra; updated following 1.2.2 release
+    max_element_length = 6
+    max_atomtype_length = 20
+
+    # Sort to make sure the atom order matches the topology indexing
+    sorted_sites = [site for site in top.sites]
+    sorted_sites.sort(key=lambda site: top.get_index(site))
+    names = [site.name for site in sorted_sites]
+    types = [site.atom_type.name for site in sorted_sites]
 
     # Check constraints on atom type length and element name length
-    # TODO: Update these following Cassandra release
-    # to be more reasonable values
     n_unique_names = len(set(names))
     for name in names:
-        if len(name) > 2:
+        if len(name) > max_element_length:
             warnings.warn(
-                "Warning, name {} will be shortened "
-                "to two characters. Please confirm your final "
-                "MCF.".format(name)
+                f"Name: {name} will be shortened to {max_element_length}"
+                "characters. Please confirm your final MCF."
             )
 
     # Confirm that shortening names to two characters does not
     # cause two previously unique atom names to become identical.
-    names = [name[:2] for name in names]
+    names = [name[:max_element_length] for name in names]
     if len(set(names)) < n_unique_names:
         warnings.warn(
-            "Warning, the number of unique names has been "
-            "reduced due to shortening the name to two "
-            "characters."
+            "The number of unique names has been reduced due"
+            f"to shortening the name to {max_element_length} characters."
         )
 
     n_unique_types = len(set(types))
-    for itype in types:
-        if len(itype) > 6:
+    for type_ in types:
```

Comment:
Here's how you use the potential filter.
And also, I think you only need to iterate over unique types here, not all types.
```python
from gmso.core.views import PotentialFilters
pfilter = PotentialFilters.UNIQUE_SORTED_NAMES

unique_typesList = top.atom_types(filter_by=pfilter))
for type_ in unique_typesList:
```

I also wouldn't save this to something called `types`. Maybe something like `atypesList` is better. Or `formatted_types`.